listenbrainz/db/user_timeline_event.py

```diff
@@ -170,6 +183,40 @@ def get_recording_recommendation_events_for_feed(user_ids: List[int], min_ts: in
         return [UserTimelineEvent(**row) for row in result.fetchall()]
 
 
+def get_personal_recommendation_events_for_feed(user_id: int, min_ts: int, max_ts: int, count: int) -> List[UserTimelineEvent]:
+    """ Gets a list of personal_recording_recommendation events for specified users.
+
+    user_ids is a tuple of user row IDs.
+    """
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT user_timeline_event.id
+                 , user_timeline_event.user_id
+                 , user_timeline_event.event_type
+                 , user_timeline_event.metadata
+                 , user_timeline_event.created
+                 , "user".musicbrainz_id as user_name
+              FROM user_timeline_event
+              JOIN "user"
+                ON user_timeline_event.user_id = "user".id
+             WHERE (user_timeline_event.metadata ->> 'followers')::jsonb @> (:user_id)::text::jsonb
```

Comment:
AFAIK, `(user_timeline_event.metadata ->> 'followers')::jsonb` should be equivalent to `user_timeline_event.metadata -> 'followers'`. Notice one less arrow. `->` keeps the output as jsonb whereas `->>` converts it to text. Since we want jsonb here anyway use the former and avoid the cast. Also, no need to cast user_id to jsonb, just casting to text is sufficient.

See https://stackoverflow.com/a/46206119 for reference.